### PR TITLE
Replace repetitive actions with Action Groups in NewProductsListWidgetTest

### LINKED ISSUE #32740: [Issue] Replace repetitive actions with Action Groups in NewProductsListWidgetTest
This issue is automatically created based on existing pull request: magento/magento2#32702: Replace repetitive actions with Action Groups in NewProductsListWidgetTest

---------

### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
Run the test in the local environment and verified that it run successfully

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by kate-kyzyma**
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
Run the test in the local environment and verified that it run successfully

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#32740: Replace repetitive actions with Action Groups in NewProductsListWidgetTest

### CODE REVIEW

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Comment by kate-kyzyma** [comment]
@magento run Functional Tests EE

File: app/code/Magento/Cms/Test/Mftf/ActionGroup/AdminExpandContentSectionActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminExpandContentSectionActionGroup">
+        <annotations>
+            <description>Expand Content section on the Admin CMS Page creation/edit.</description>
+        </annotations>
+        <click selector="{{CmsNewPagePageContentSection.header}}" stepKey="expandContentSection"/>
+        <waitForPageLoad stepKey="waitForSeoSection"/>
```
Comment (eduard13) [inline app/code/Magento/Cms/Test/Mftf/ActionGroup/AdminExpandContentSectionActionGroup.xml]:
```suggestion
        <waitForPageLoad stepKey="waitForContentSection"/>
```

File: app/code/Magento/Cms/Test/Mftf/ActionGroup/AdminExpandSeoSectionActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminExpandSeoSectionActionGroup">
+        <annotations>
+            <description>Expand SEO section on the Admin CMS Page creation/edit.</description>
+        </annotations>
+        <click selector="{{CmsNewPagePageSeoSection.header}}" stepKey="expandSeoSection"/>
+        <waitForPageLoad stepKey="waitForContentSection"/>
```
Comment (eduard13) [inline app/code/Magento/Cms/Test/Mftf/ActionGroup/AdminExpandSeoSectionActionGroup.xml]:
```suggestion
        <waitForPageLoad stepKey="waitForSeoSection"/>
```

File: app/code/Magento/Widget/Test/Mftf/Test/NewProductsListWidgetTest.xml [hunk 3]
```diff
@@ -34,18 +34,25 @@
...
+        <comment userInput="Comment is added to preserve the step key for backward compatibility" stepKey="fillNoOfProductToDisplay"/>
         <actionGroup ref="AdminClickInsertWidgetActionGroup" stepKey="clickInsertWidget"/>
-        <click selector="{{CmsNewPagePageSeoSection.header}}" stepKey="expandSeoSection"/>
-        <fillField selector="{{CmsNewPagePageSeoSection.urlKey}}" userInput="{{_newDefaultCmsPage.identifier}}" stepKey="fillPageUrlKey"/>
-        <click selector="{{CmsNewPagePageActionsSection.saveAndContinueEdit}}" stepKey="clickSaveCmsPage"/>
+        <actionGroup ref="AdminExpandSeoSectionActionGroup" stepKey="expandSeoSection"/>
+        <actionGroup ref="AdminCmsPageSetUrlActionGroup" stepKey="fillPageUrlKey">
+            <argument name="urlKey" value="{{_newDefaultCmsPage.identifier}}"/>
+        </actionGroup>
+        <actionGroup ref="SaveCmsPageActionGroup" stepKey="clickSaveCmsPage" />
```
Comment (eduard13) [inline app/code/Magento/Widget/Test/Mftf/Test/NewProductsListWidgetTest.xml]:
```suggestion
        <actionGroup ref="SaveCmsPageActionGroup" stepKey="clickSaveCmsPage"/>
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @kate-kyzyma, could you please check the minor suggestions below?

Thank you for your input.

**Comment by eduard13** [comment]
@magento create issue

**Comment by kate-kyzyma** [comment]
Hi @eduard13!

I have fixed mentioned issues. Please, check.

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Pull request closed without merging**